streams/src/main/java/org/apache/kafka/streams/processor/internals/ChangelogReader.java

```diff
@@ -21,29 +21,35 @@
 
 import java.util.Collection;
 import java.util.Map;
+import java.util.Set;
 
 /**
  * Performs bulk read operations from a set of partitions. Used to
- * restore  {@link org.apache.kafka.streams.processor.StateStore}s from their
- * changelogs
+ * restore a state store from {@link TopicPartition} changelog.
  */
 public interface ChangelogReader {
     /**
      * Register a state store and it's partition for later restoration.
-     * @param restorer the state restorer to register
+     *
+     * @param partition the changelog topic partition to restore
+     * @param stateManager the state manager used for restoring (one per task)
      */
-    void register(final StateRestorer restorer);
+    void register(final TopicPartition partition, final ProcessorStateManager stateManager);
 
     /**
      * Restore all registered state stores by reading from their changelogs.
-     * @return all topic partitions that have been restored
      */
-    Collection<TopicPartition> restore(final RestoringTasks active);
+    void restore();
```

Comment:
We do not return the partitions that have been restored, instead it is queryable from `completedChangelogs` below, which is cleaner.